src/start.sh

```diff
@@ -75,33 +97,62 @@ start() {
     # Start tailing the FTL log from the most recent "FTL Started" message
     # Get the line number
     startFrom=$(grep -n '########## FTL started' /var/log/pihole/FTL.log | tail -1 | cut -d: -f1)
-    # Start the tail from the line number
-    tail -f -n +${startFrom} /var/log/pihole/FTL.log &
+    # Start the tail from the line number and background it
+    tail --follow=name -n +${startFrom} /var/log/pihole/FTL.log &
   else
     echo "  [i] FTL log output is disabled. Remove the Environment variable TAIL_FTL_LOG, or set it to 1 to enable FTL log output."
   fi
 
-  # https://stackoverflow.com/a/49511035
-  wait $!
+  # Wait for the ftl process to finish and handle its return
+  wait $ftl_pid
+  stop $?
 }
 
 stop() {
-  # Ensure pihole-FTL shuts down cleanly on SIGTERM/SIGINT
-  ftl_pid=$(pgrep pihole-FTL)
-  killall --signal 15 pihole-FTL
+  local FTL_EXIT_CODE=$1
 
-  # Wait for pihole-FTL to exit
-  while test -d /proc/"${ftl_pid}"; do
-    sleep 0.5
-  done
+  # Only attempt to close pihole-FTL if it is running, it may already have crashed
+  if pgrep pihole-FTL >/dev/null; then
+    echo ""
+    echo "  [i] Container stop requested..."
+    echo "  [i] pihole-FTL is running - Attempting to shut it down cleanly"
+    echo ""
+    # Ensure pihole-FTL shuts down cleanly on SIGTERM/SIGINT
+    ftl_pid=$(pgrep pihole-FTL)
+
+    killall --signal 15 pihole-FTL
+
+    # Wait for pihole-FTL to exit
+    while test -d /proc/"${ftl_pid}"; do
+      sleep 0.5
+    done
+    # Return from this function, it will be called again once FTL finishes
+    return
+  fi
 
-  # If we are running pytest, keep the container alive for a little longer
-  # to allow the tests to complete
+  # Wait for a few seconds to allow the FTL log tail to catch up before exiting the container
+  sleep 2
+
+  # ensure the exit code is an integer, if not set it to 1
+  if ! [[ "${FTL_EXIT_CODE}" =~ ^[0-9]+$ ]]; then
+    FTL_EXIT_CODE=1
+  fi
+
+  echo ""
+  echo "  [i] pihole-FTL exited with status $FTL_EXIT_CODE"
+  echo ""
+  echo "  [i] Container will now stop or restart depending on your restart policy"
+  echo "      https://docs.docker.com/engine/containers/start-containers-automatically/#use-a-restart-policy"
+  echo ""
+
+  # # If we are running pytest, keep the container alive for a little longer
+  # # to allow the tests to complete
```

Comment:
Did you add the extra `#` on purpose?

```suggestion
  # If we are running pytest, keep the container alive for a little longer
  # to allow the tests to complete
```

Reply:
ha, no. Copy paste issues while I was moving things around most likely